Add ocean resolution to setup_expt invocation and retire/reduce COMROT/ROTDIR usage

KateFriedman-NOAA opened the pull request:
# Description

This PR combines two issues, both of which make updates to the global-workflow setup API.

Firstly, this PR does some housekeeping for issue #2061:

1. Removes the `COMROT` variable, replaces it with `ROTDIR` or other appropriate variable(s), and reduces the usage of `ROTDIR` where able. Some instances of `ROTDIR` remain since they are necessary still. Some updates were made to scripts that may be removed or rewritten in the future; changes were still made for consistency and to remove `COMROT`/`ROTDIR`.
2. The hosts files are updated to remove `COMROOT` and move the values into the `COMINsyn` settings (which is where they were used anyway). Need to free up `COMROOT` to be filled by the `--comroot` setup argument.
3. Additionally, several instances of `COMROT` were changed to be `COMROOT` or `comroot` (2nd "O"/"o" added). The use of `--comroot` as an argument for the setup scripts and `COMROOT` to define `ROTDIR` in `config.base` brings the system more in line with how NCO uses `COMROOT` in operations to define the location of com (similar to where developers place their `ROTDIR`s. This is a subtle and easy to miss update for the setup script arguments (`comrot` -> `comroot`). An announcement is recommended so users begin using `--comroot` instead of `--comrot` with the setup scripts.
4. Documentation is also updated

Secondly, this PR allows a user to provide a desired ocean resolution during the invocation of setup_expt.py:

1. renames `--resdet` option to `--resdetatmos`
2. adds `--resdetocean`. If a value `<= 0`. is provided, the program will choose based on pre-selected options.
3. updates the CI test yamls
5. updates the documentation

Assigning @TerrenceMcGuinness-NOAA as a reviewer since this PR touches the CI files.

Resolves #2061

This PR has been updated to merge PR #2204 into it.

# Type of change

- New feature (adds functionality)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? No, but users will need to update their use of setup_expt.py
- Does this change require a documentation update? YES

# How has this been tested?

- Created a few of the experiment directories using `create_experiment.py` and the various `ci/cases` yaml files and inspected `config.base` in the respective experiment directories.
- Cycled atmos-only C96C48 test on Hera.
- Forecast-only S2SW C48 test on Hera.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

KateFriedman-NOAA committed:
Update ROTDIR references in documentation

- Remove references to COMROT and replace with ROTDIR

Refs #2061

KateFriedman-NOAA committed:
Update hosts files to remove COMROOT

- Remove COMROOT setting for each platform
- Replace COMROOT with full path in COMINsyn

Refs #2061

KateFriedman-NOAA committed:
Replace COMROT with ROTDIR in comment

Refs #2061

KateFriedman-NOAA committed:
Replace COMROT/comrot with ROTDIR/comroot in setup

- Remove instances of COMROT/comrot in setup scripts
- Replace with ROTDIR or comroot as appropriate

Refs #2061

KateFriedman-NOAA committed:
Replace rotdir with comroot in README

Refs #2061

KateFriedman-NOAA committed:
Update ROTDIR usage in configs

- Replace ROTDIR with COMROOT in base configs for GFS and GEFS
- Update function name in config.com

Refs #2061

KateFriedman-NOAA committed:
Replace COMROT with ROTDIR in ci system

Refs #2061

aerorahul committed:
add option to provide an ocean resolution during setup_expt.py

aerorahul committed:
Merge branch 'develop' into feature/setup_expt_oceanres

aerorahul committed:
Merge branch 'develop' into feature/setup_expt_oceanres

aerorahul committed:
fix pynorm error

aerorahul committed:
Merge branch 'develop' into feature/setup_expt_oceanres

aerorahul committed:
Merge branch 'develop' into feature/setup_expt_oceanres

aerorahul committed:
update yamls for ci tests

aerorahul committed:
update documentation instances of resdet and resens; add resdetocean

aerorahul committed:
update default resdetocean

aerorahul committed:
Merge branch 'develop' into feature/setup_expt_oceanres

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/clean_out_rotdir

KateFriedman-NOAA committed:
Updated some scripts for COM directories

Updates in some JJOB and ex-scripts remove use of ROTDIR
and replace with appropriate COM_* directories.

Refs #2061

KateFriedman-NOAA commented:
The forecast-only S2SW C48 test on Hera is still running at the time of initiating this PR. The IC and prep jobs have completed successfully though.

WalterKolczynski-NOAA commented on jobs/JGFS_ATMOS_FSU_GENESIS:
```diff
@@ -33,25 +33,18 @@ export PYTHONPATH=${USHens_tracker}/FSUgenesisPY:${PYTHONPATH}
+export COMIN=${COM_ATMOS_GRIB_0p25}
+export gfsdir=${COM_ATMOS_GRIB_0p25}
+export COMINgfs=${COM_ATMOS_GRIB_0p25}
+
```
- Does the genesis script use all of `COMIN`, `gfsdir`, and `COMINgfs`?
- `gdasdir` (and `COMINgdas`) are no longer being defined.

KateFriedman-NOAA replied:
> * Does the genesis script use all of `COMIN`, `gfsdir`, and `COMINgfs`?

Yes, there is a mixture of all three across both the `TC_tracker` (dev) and `ens_tracker` (ops) packages. Note, the FSU_GENESIS job is not tested so further changes to the JJOB script are potentially needed.

> * `gdasdir` (and `COMINgdas`) are no longer being defined.

The FSU genesis job is only run for `gfs` in operations so we don't need the `gdas` versions of those path variables.

WalterKolczynski-NOAA commented on jobs/JGLOBAL_ATMOS_POST_MANAGER:
```diff
@@ -32,10 +26,8 @@ export USHgfs=${HOMEgfs:-${HOMEgfs}/ush}
-export COMOUT=${COMOUT:-${ROTDIR}/${RUN}.${PDY}/${cyc}/atmos}
-
+YMD=${PDY} HH=${cyc} generate_com -rx COM_ATMOS_HISTORY
+YMD=${PDY} HH=${cyc} generate_com -rx COM_ATMOS_GRIB
```
I don't think this second one is needed
```suggestion
```

KateFriedman-NOAA committed:
Update jobs/JGLOBAL_ATMOS_POST_MANAGER

Remove unneeded `generate_com` command.

Refs #2061

Co-authored-by: Walter Kolczynski - NOAA <Walter.Kolczynski@noaa.gov>

KateFriedman-NOAA commented:
> create_experiment.py: error: unrecognized arguments: --rotdir /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2214/RUNTESTS/ROTDIR
> ROOT/PR/2214/RUNTESTS/ROTDIR --expdir /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2214/RUNTESTS/EXPDIR --idate 2021032312 --edate 2021032312 --yaml /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2214/global-workflow/ci/platforms/gfs_defaults_ci.yaml
>

I see the problem and how to fix. Will fix and retry CI tests.

aerorahul committed:
Update docs/doxygen/mainpage.h

KateFriedman-NOAA committed:
Change rotdir/ROTDIR to comroot/COMROOT in CI tests

The comrot/COMROT variables should have been changed to
comroot/COMROOT (with 2nd o/O) instead of rotdir/ROTDIR.
Correcting now.

Refs #2061

KateFriedman-NOAA committed:
Change ROTDIR to COMROOT in CI check script

Refs #2061

TerrenceMcGuinness-NOAA reviewed (approved):
Nice work, so cool.  So glad to get rid of the ROT stuff!  Looking forward to studying the updated documentation to finally get a handle on the staging directories.

KateFriedman-NOAA commented:
Killed CI tests again after receiving request to merge branch from PR #2204 into branch for this PR. Will merge branches/PRs and then restart CI tests.

KateFriedman-NOAA committed:
Merge PR #2204 branch into PR #2214 branch

Merge remote-tracking branch 'upstream/feature/setup_expt_oceanres' into feature/clean_out_rotdir

* upstream/feature/setup_expt_oceanres:
  Update docs/doxygen/mainpage.h
  update default resdetocean
  update documentation instances of resdet and resens; add resdetocean
  update yamls for ci tests
  fix pynorm error
  add option to provide an ocean resolution during setup_expt.py

aerorahul committed:
Update ci/cases/pr/C96_atm3DVar.yaml

KateFriedman-NOAA committed:
Merge remote-tracking branch 'upstream/feature/setup_expt_oceanres' into feature/clean_out_rotdir

* upstream/feature/setup_expt_oceanres:
  Update ci/cases/pr/C96_atm3DVar.yaml

Refs #2061

WalterKolczynski-NOAA reviewed (approved):
Conditionally approved pending successful completion of CI testing.

TerrenceMcGuinness-NOAA reviewed (approved):
changes to CI scripts look great.  All substantive updates in the CI Bash scrips pertain to creating experiment which is ran via the PR during CI run-time so no updates to the In-Place cron scripts are needed to test.

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/clean_out_rotdir
